### PR TITLE
feat(auth): add minitailor authentication strategy

### PR DESCRIPTION
**Pull request by IzumiSy**
# Background

Since minitailor supports fake authentication provider, so auth package should also support that.

# Changes

* Add `MinitailorStrategy` as a new authentication strategy
* Fix a bug in manual-callback strategy

### CODE REVIEW

File: packages/auth/src/client/hooks.ts [hunk 1]
```diff
@@ -50,11 +50,12 @@ export const useAuth = () => {
       case "redirection":
         window.location.replace(result.uri);
         break;
-      case "manual-callback":
-        await fetch(config.loginCallbackPath(strategy.name()), {
-          body: result.payload,
-        });
```
Comment (IzumiSy) [inline packages/auth/src/client/hooks.ts:56]: Callback works as page rendering, so calling middlewares with fetch API is not what it should be.

File: packages/auth/src/core/config.ts [hunk 2]
```diff
@@ -63,8 +65,9 @@ export class Config {
     return this.params.unauthorizedPath || "/unauthorized";
   }
 
-  loginCallbackPath(strategy: string = defaultStrategy.name()) {
-    return this.params.loginCallbackPath || `/login/callback/${strategy}`;
```
Comment (IzumiSy) [inline packages/auth/src/core/config.ts:67]: This always fallbacks `default` strategy, but it should not be.

File: packages/auth/README.md [hunk 3]
```diff
@@ -111,24 +128,33 @@ import { useAuth } from "@tailor-platform/auth/client";
 // Redirect to `/dashboard` after logging in
 const Component = async () => {
   const { login } = useAuth();
+  const doLogin = useCallback(() => {
+    login({
+      // name is optional, but you can specify one of the following:
+      // * oidc (default)
+      // * saml
+      // * minitailor
```
Comment (IzumiSy) [inline packages/auth/README.md:136]: With this PR, `minitailor` here would be available to users.

**Comment by masato-io** [comment]
looks like lint check is failing on CI

**Review by masato-io — APPROVED** [review]

File: packages/auth/src/core/config.ts [hunk 4]
```diff
@@ -63,8 +65,9 @@ export class Config {
...
+  loginCallbackPath(strategy?: string) {
+    const basePath = this.params.loginCallbackPath || `/login/callback`;
+    return strategy ? basePath + "/" + strategy : basePath;
```
Comment (jfeltesse) [inline packages/auth/src/core/config.ts:70]: IMHO it would be more readable doing

```
    return strategy ? `${basePath}/${strategy}` : basePath;
```

Comment (IzumiSy) [reply packages/auth/src/core/config.ts]: Agreed 👍  Updated.

**Pull request merged**